client/state/posts/middleware.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * External dependencies
+ */
+import { cloneDeep, isNumber, map } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { TERMS_RECEIVE } from 'state/action-types';
+import { getEditedPost } from 'state/posts/selectors';
+import { editPost } from 'state/posts/actions';
+
+/**
+ * Dispatches editPost when a new term has been added
+ * that also has a postId on the action
+ *
+ * @param  {Function} dispatch  Dispatch method
+ * @param  {Object}   state     Global state tree
+ * @param  {Object}   action    Action object
+ */
+export function onTermsReceive( dispatch, state, action ) {
+	const { postId, taxonomy, terms, siteId } = action;
+	const post = getEditedPost( state, siteId, postId );
+	const newTerm = terms[ 0 ];
+
+	// if there is no post, no term, or term is temporary, bail
+	if ( ! post || ! newTerm || ! isNumber( newTerm.ID ) ) {
+		return;
+	}
+
+	const postTerms = cloneDeep( post.terms ) || {};
```

Comment:
>Could likely get away with an `Object.assign`

I'm not sure you even need that. You don't seem to be mutating `postTerms` at all after this line.